spec.emu

```diff
@@ -182,6 +182,21 @@ contributors: Ujjwal Sharma, Younies Mahmoud
           1. Assert: 𝔽(_v_) is finite.
           1. If _v_ &lt; 0 and _sign_ &gt; 0, return *false*.
           1. If _v_ &gt; 0 and _sign_ &lt; 0, return *false*.
+        1. Let _years_ be _record_.[[Years]].
+        1. Let _months_ be _record_.[[Months]].
+        1. Let _weeks_ be _record_.[[Weeks]].
+        1. If abs(_years_) &ge; 2<sup>32</sup>, return *false*.
+        1. If abs(_months_) &ge; 2<sup>32</sup>, return *false*.
+        1. If abs(_weeks_) &ge; 2<sup>32</sup>, return *false*.
+        1. Let _days_ be _record_.[[Days]].
+        1. Let _hours_ be _record_.[[Hours]].
+        1. Let _minutes_ be _record_.[[Minutes]].
+        1. Let _seconds_ be _record_.[[Seconds]].
+        1. Let _milliseconds_ be _record_.[[Milliseconds]].
+        1. Let _microseconds_ be _record_.[[Microseconds]].
+        1. Let _nanoseconds_ be _record_.[[Nanoseconds]].
+        1. Let _normalizedSeconds_ be _days_ &times; 86,400 + _hours_ &times; 3600 + _minutes_ &times; 60 + _seconds_ + _milliseconds_ &times; 10<sup>-3</sup> + _microseconds_ &times; 10<sup>-6</sup> + _nanoseconds_ &times; 10<sup>-9</sup>.
```

Comment:
Question: how is this check performed in Temporal? It seems like it could be expensive to perform this check since it requires performing arithmetic in BigDecimal space (since these are presumably MVs).

Reply:
In Temporal it's performed the same way. You don't need big decimals to implement it, because the exact value doesn't matter if it's over the limit. With JS Numbers, for example, you can implement it as
```js
Number.isSafeInteger(d * 86400 + h * 3600 + min * 60 + s + Math.trunc(ms / 1e3 + µs / 1e6 + ns / 1e9))
```

Reply:
Anba [showed](https://github.com/tc39/proposal-temporal/pull/2727#discussion_r1399142661) me how my statement above was incorrect. However, I believe it's still possible to perform the check without using big decimals, and I've rewritten the relevant section of the Temporal spec text to do the calculation in a way that's more likely to be correct if naively followed using float arithmetic.

https://github.com/tc39/proposal-temporal/pull/2727/commits/a9abc12527a46106d54ee060b6fa91704ce0a379

I'd recommend rewriting this text in the same way (at least, if you agree that I did this correctly!)